author	Peter Tsao <peter.tsao@mediatek.com>	2024-04-15 22:19:22 +0800
committer	Luiz Augusto von Dentz <luiz.von.dentz@intel.com>	2024-05-14 10:51:05 -0400
commit	958cd6beab693f395ffe07814ef77d2b45e8b0fc (patch)
tree	a8912c0d1dda9349ff6e59d19a7ffba0c260f8c2
parent	68aa21054ec3a1a313af90a5f95ade16c3326d20 (diff)
download	linux-next-958cd6beab693f395ffe07814ef77d2b45e8b0fc.tar.gz
Bluetooth: btusb: Fix the patch for MT7920 the affected to MT7921
Because both MT7920 and MT7921 use the same chip ID.
We use the 8th bit of fw_flavor to distingush MT7920.
The original patch made a mistake to check whole fw_flavor,
that makes the condition both true (dev_id == 0x7961 && fw_flavor),
and makes MT7921 flow wrong.

In this patch, we correct the flow to get the 8th bit value for MT7920.
And the patch is verified pass with both MT7920 and MT7921.

Signed-off-by: Peter Tsao <peter.tsao@mediatek.com>
Signed-off-by: Luiz Augusto von Dentz <luiz.von.dentz@intel.com>
Diffstat
-rw-r--r--	drivers/bluetooth/btusb.c	1	
1 files changed, 1 insertions, 0 deletions
diff --git a/drivers/bluetooth/btusb.c b/drivers/bluetooth/btusb.c
index 4c0cc13adb47dc..8bede0a335668a 100644
--- a/drivers/bluetooth/btusb.c
+++ b/drivers/bluetooth/btusb.c
@@ -3123,6 +3123,7 @@ static int btusb_mtk_setup(struct hci_dev *hdev)
 			bt_dev_err(hdev, "Failed to get fw flavor (%d)", err);
 			return err;
 		}
+		fw_flavor = (fw_flavor & 0x00000080) >> 7;
 	}
 
 	mediatek = hci_get_priv(hdev);
